src/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapAsyncResult.cs

```diff
@@ -2,97 +2,70 @@
+
 namespace System.DirectoryServices.Protocols
 {
-    using System;
-    using System.Threading;
-    using System.Net;
-    using System.Text;
-    using System.IO;
-    using Microsoft.Win32.SafeHandles;
-
     internal class LdapAsyncResult : IAsyncResult
     {
         private LdapAsyncWaitHandle _asyncWaitHandle = null;
-        internal AsyncCallback callback = null;
-        internal bool completed = false;
+        internal AsyncCallback _callback = null;
+        internal bool _completed = false;
         private bool _completedSynchronously = false;
-        internal ManualResetEvent manualResetEvent = null;
+        internal ManualResetEvent _manualResetEvent = null;
         private object _stateObject = null;
-        internal LdapRequestState resultObject = null;
-        internal bool partialResults = false;
+        internal LdapRequestState _resultObject = null;
+        internal bool _partialResults = false;
 
         public LdapAsyncResult(AsyncCallback callbackRoutine, object state, bool partialResults)
         {
             _stateObject = state;
-            callback = callbackRoutine;
-            manualResetEvent = new ManualResetEvent(false);
-            this.partialResults = partialResults;
+            _callback = callbackRoutine;
+            _manualResetEvent = new ManualResetEvent(false);
+            _partialResults = partialResults;
         }
 
-        object IAsyncResult.AsyncState
-        {
-            get { return _stateObject; }
-        }
+        object IAsyncResult.AsyncState => _stateObject;
 
         WaitHandle IAsyncResult.AsyncWaitHandle
         {
-            get
-            {
-                if (null == _asyncWaitHandle)
-                {
-                    _asyncWaitHandle = new LdapAsyncWaitHandle(manualResetEvent.SafeWaitHandle);
-                }
-
-                return (WaitHandle)_asyncWaitHandle;
-            }
+            get => _asyncWaitHandle ?? (_asyncWaitHandle = new LdapAsyncWaitHandle(_manualResetEvent.SafeWaitHandle));
         }
 
-        bool IAsyncResult.CompletedSynchronously
-        {
-            get { return _completedSynchronously; }
-        }
+        bool IAsyncResult.CompletedSynchronously => _completedSynchronously;
 
-        bool IAsyncResult.IsCompleted
-        {
-            get { return completed; }
-        }
+        bool IAsyncResult.IsCompleted => _completed;
 
-        public override int GetHashCode()
-        {
-            return manualResetEvent.GetHashCode();
-        }
+        public override int GetHashCode() => _manualResetEvent.GetHashCode();
 
-        public override bool Equals(object o)
+        public override bool Equals(object obj)
         {
-            if ((!(o is LdapAsyncResult)) || (o == null))
+            if (!(obj is LdapAsyncResult otherAsyncResult))
             {
                 return false;
             }
 
-            return (this == (LdapAsyncResult)o);
+            return this == otherAsyncResult;
```

Comment:
This method is strange.  I don't know why the override is necessary or why it's not just `public override bool Equals(object obj) => base.Equals(obj);`